std/repl.glu

```diff
@@ -94,14 +94,14 @@ let loop x: () -> IO () =
                 >>= io.print
                 >> return True
 
-    repl_prim.input "> " >>= \line_opt ->
+    rustyline.readline editor "> " >>= \line_opt ->
         match line_opt with
             | None -> return ()
             | Some line -> run_line line >>= \continue ->
-                if continue then loop () else return ()
+                if continue then loop editor else return ()
 
 let run x: () -> IO () =
     io.print "gluon (:h for help, :q to quit)"
-        >>= loop
+        >>= \_-> loop (rustyline.new_editor ())
```

Comment:
Yes it should. I may be a bit averse to using >> since it may actually
change the semantics of a program since gluon isn't pure but in this
instance I should have used >>

On 18 Jul 2016 17:29, "Brendan Zabarauskas" <notifications@github.com>
wrote:
>
> In std/repl.glu:
>
> > > let run x: () -> IO () = > io.print "gluon (:h for help, :q to quit)"
> - >>= loop > + >>= \_-> loop (rustyline.new_editor ())
>
> Does (>>) work here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.